### PR TITLE
Unit tests 2ndorder

### PR DESCRIPTION
**Pull request by snafees**
Renamed files/dirs accordingly and added tests to check 2nd order work. Tests pass!

### CODE REVIEW

File: ortho_seq_code/conftest.py [hunk 1]
```diff
@@ -49,28 +47,40 @@ def protein_nopad_expected_output_dir(protein_expected_output_dir):
...
 
 @pytest.fixture
-def nucleotide_expected_output_dir(nucleotide_two_sites_data_dir):
-    return os.path.join(nucleotide_two_sites_data_dir, "expected_outputs")
+def nucleotide_data_dir(test_data_dir):
+    return os.path.join(test_data_dir, "nucleotide")
 
+# @pytest.fixture
+# def nucleotide_two_sites_data_dir(nucleotide_data_dir):
+#     return os.path.join(nucleotide_data_dir, "second_order")
```
Comment (phoenixAja) [inline ortho_seq_code/conftest.py:55]: remove comments:
```suggestion
```

File: ortho_seq_code/conftest.py [hunk 2]
```diff
@@ -49,28 +47,40 @@ def protein_nopad_expected_output_dir(protein_expected_output_dir):
...
 
 @pytest.fixture
-def protein_pheno_no_padding(protein_data_dir):
-    return os.path.join(protein_data_dir, 'protein_pheno_nopad.txt')
+def nucleotide_second_order_expected_output_dir(nucleotide_second_order_data_dir):
+    return os.path.join(nucleotide_second_order_data_dir, "expected_outputs")
 
+# @pytest.fixture
+# def nucleotide_expected_output_dir(nucleotide_two_sites_data_dir):
+#     return os.path.join(nucleotide_two_sites_data_dir, "expected_outputs")
```
Comment (phoenixAja) [inline ortho_seq_code/conftest.py:74]: remove comments:
```suggestion
```

**Review by phoenixAja — APPROVED** [review]
nice work! I like how you got rid of a lot of duplicate code in orthogonal_ploynomial by using existing variables 👍

**Comment by snafees** [comment]
> nice work! I like how you got rid of a lot of duplicate code in orthogonal_ploynomial by using existing variables 👍

Yes for sure! I had been meaning to do that! Ah! Finally!

File: ortho_seq_code/tests/test_orthoseqs.py [hunk 3]
```diff
@@ -23,21 +23,63 @@ def output_file_templates(basename):
         f"{basename}_reg11i1.npy",
         f"{basename}_Pa1i1.npy",
         f"{basename}_P1D.npy",
-        f"{basename}_varP1D.npy",
+        f"{basename}_varP1D.npy", #end of building first order space
+
+    ]
+
+def output_file_2ndorder_templates(basename):
```
Comment (pranathivemuri) [inline ortho_seq_code/tests/test_orthoseqs.py:30]: if you use saving all the arrays as npz format you would not have to do this also you should be using refex to see if you have 2 in the string or something like that  if its 2nd order

File: ortho_seq_code/orthogonal_polynomial.py [hunk 4]
```diff
@@ -876,8 +878,12 @@ def orthogonal_polynomial(filename, pheno_file, molecule, sites, dm, pop_size, p
                             for m in range(dm):
                                 var2D[j][k][l][m] += (P2D[j][k][i][l][m] ** 2) / pop_size
         print("computed var2D")
-        naming = os.path.basename(f.name)
-        np.save(os.path.join(out_dir, naming + str('_var2D')), var2D)
+        arrays_save[naming + '_var2D'] = var2D
+        #np.save(os.path.join(out_dir, naming + str('_var2D')), var2D)
+
```
Comment (pranathivemuri) [inline ortho_seq_code/orthogonal_polynomial.py:883]: This np.save should be removed

**Comment by pranathivemuri** [comment]
https://github.com/snafees/ortho_seqs/pull/13 closing in favor of

**Pull request closed without merging**